### PR TITLE
Check connection state before sending gRPC request

### PR DESCRIPTION
**Pull request by peterbarrow**
We are seeing a problem where the gRPC calls are hanging for 15 or so minutes when trying to reconnect to a node. This might be due to us filling the send buffer with requests while the connection is down. I have added code to check if the connection is ready to send before we send any more requests.

### CODE REVIEW

File: bot/normal/normal.go [hunk 1]
```diff
@@ -449,22 +449,22 @@ func (b *Bot) initialiseData() error {
 
 	err = b.lookupInitialValues()
 	if err != nil {
-		b.log.Debug("Stopping position management as we could not get initial values")
+		b.log.Debugf("Stopping position management as we could not get initial values: %v", err)
```
Comment (ashleyvega) [inline bot/normal/normal.go:452]: I hate to be picky, but could you use:
```go
log.WithFields(log.Fields{
	"error": err.Error(),
}).Debug("a fixed string")
```

File: bot/normal/normal.go [hunk 2]
```diff
@@ -485,14 +485,14 @@ func (b *Bot) placeAuctionOrders() {
 
 	// Place the random orders split into
 	var totalVolume uint64
-	r := rand.New(rand.NewSource(1)) // #nosec G404 This suboptimal rand generator is fine for now
-
+	rand.Seed(time.Now().UnixNano())
+	/* #nosec G404 */
```
Comment (ashleyvega) [inline bot/normal/normal.go:489]: is this `nosec` comment still needed?

File: node/grpcnode.go [hunk 3]
```diff
@@ -52,7 +53,7 @@ func (n *GRPCNode) GetAddress() (url.URL, error) {
 // SubmitTransaction submits a signed transaction
 func (n *GRPCNode) SubmitTransaction(req *api.SubmitTransactionRequest) (response *api.SubmitTransactionResponse, err error) {
 	msg := "gRPC call failed: SubmitTransaction: %w"
-	if n == nil {
+	if n == nil || n.conn.GetState() != connectivity.Ready {
```
Comment (ashleyvega) [inline node/grpcnode.go:56]: I hate to be picky, but could we differentiate between `n == nil` and `n.conn.GetState() != connectivity.Ready`?

Perhaps a new `e.NotReady` containing "gRPC connection not ready" or similar.

**Review by ashleyvega — CHANGES_REQUESTED** [review]
Fingers crossed this helps with slow reconnections.

But in the meantime, changes requested, please.

File: bot/normal/normal.go (hunk 1 above)
Comment (peterbarrow) [reply bot/normal/normal.go]: Updated all I could find.

File: node/grpcnode.go (hunk 3 above)
Comment (peterbarrow) [reply node/grpcnode.go]: Done, thanks.

File: bot/normal/normal.go (hunk 2 above)
Comment (ashleyvega) [reply bot/normal/normal.go:492]: ok

File: bot/normal/normal.go (hunk 1 above)
Comment (ashleyvega) [reply bot/normal/normal.go]: thanks

**Review by ashleyvega — APPROVED** [review]
Thanks for making changes, LGTM.

**Pull request merged**